### PR TITLE
Osxuniversal leak check

### PR DESCRIPTION
**Pull request by kpanwar**
leak cleanups
<a href='#crh-start'></a><a href='#crh-data-%7B%22approved%22%3A%20%7B%22https%3A//github.com/RPangrle-MS%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7004783%3Fv%3D3%22%7D%7D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-objc/pull/309%23issuecomment-98269701%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%3Ashipit%3A%22%2C%20%22created_at%22%3A%20%222015-05-02T00%3A33%3A44Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7004783%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/RPangrle-MS%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%200a13f57f70ef395c3ff480b2fb63935b89a5f245%20ADAL-OSX/ADAL-OSX/ADCredentialCollectionController.m%2013%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-objc/pull/309%23discussion_r29541094%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Delete%20these%20lines%20entirely.%20We%20don%27t%20need%20them%20laying%20around.%22%2C%20%22created_at%22%3A%20%222015-05-01T23%3A33%3A02Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7004783%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/RPangrle-MS%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%22%2C%20%22created_at%22%3A%20%222015-05-02T00%3A23%3A20Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/5713915%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/kpanwar%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20ADAL-OSX/ADAL-OSX/ADCredentialCollectionController.m%3AL34-45%22%7D%2C%20%22Pull%200a13f57f70ef395c3ff480b2fb63935b89a5f245%20ADAL-OSX/ADAL-OSX/ADNTLMHandlerOSX.m%2063%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-objc/pull/309%23discussion_r29541155%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Add%20some%20SAFE_ARC_RETAINs%20to%20these%20fields%20or%20use%20%5Bself%20setUsername%3A%5D%20%26%20%5Bself%20setPassword%3A%5D%20%28or%20the%20dot%20syntax%29%22%2C%20%22created_at%22%3A%20%222015-05-01T23%3A34%3A32Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7004783%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/RPangrle-MS%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-none%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222015-05-02T00%3A25%3A25Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/5713915%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/kpanwar%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20ADAL-OSX/ADAL-OSX/ADNTLMHandlerOSX.m%3AL26-106%22%7D%2C%20%22Pull%200a13f57f70ef395c3ff480b2fb63935b89a5f245%20ADAL-OSX/ADAL-OSX/ADNTLMHandlerOSX.m%2042%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-objc/pull/309%23discussion_r29541163%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Add%20SAFE_ARC_RELEASE%20for%20_username%20and%20_password%20as%20well%22%2C%20%22created_at%22%3A%20%222015-05-01T23%3A34%3A51Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7004783%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/RPangrle-MS%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-complete%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222015-05-02T00%3A25%3A25Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/5713915%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/kpanwar%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20ADAL-OSX/ADAL-OSX/ADNTLMHandlerOSX.m%3AL26-106%22%7D%7D%2C%20%22processed%22%3A%20%5B%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-objc/pull/309%23issuecomment-98269701%22%2C%20%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-objc/pull/309%23discussion_r29541094%22%2C%20%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-objc/pull/309%23discussion_r29541155%22%2C%20%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-objc/pull/309%23discussion_r29541163%22%2C%20%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-objc/pull/309%23discussion_r29542594%22%2C%20%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-objc/pull/309%23discussion_r29542644%22%2C%20%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-objc/pull/309%23discussion_r29542645%22%5D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>

<a href='https://github.com/RPangrle-MS'></a>

- [x] <a href='#crh-comment-Pull 0a13f57f70ef395c3ff480b2fb63935b89a5f245 ADAL-OSX/ADAL-OSX/ADCredentialCollectionController.m 13'></a> &nbsp;<b><a href='https://github.com/AzureAD/azure-activedirectory-library-for-objc/pull/309#discussion_r29541094'>File: ADAL-OSX/ADAL-OSX/ADCredentialCollectionController.m:L34-45</a></b>
- <a href='https://github.com/RPangrle-MS'></a> Delete these lines entirely. We don't need them laying around.
- [x] <a href='#crh-comment-Pull 0a13f57f70ef395c3ff480b2fb63935b89a5f245 ADAL-OSX/ADAL-OSX/ADNTLMHandlerOSX.m 63'></a> &nbsp;<b><a href='https://github.com/AzureAD/azure-activedirectory-library-for-objc/pull/309#discussion_r29541155'>File: ADAL-OSX/ADAL-OSX/ADNTLMHandlerOSX.m:L26-106</a></b>
- <a href='https://github.com/RPangrle-MS'></a> Add some SAFE_ARC_RETAINs to these fields or use [self setUsername:] & [self setPassword:] (or the dot syntax)
- [x] <a href='#crh-comment-Pull 0a13f57f70ef395c3ff480b2fb63935b89a5f245 ADAL-OSX/ADAL-OSX/ADNTLMHandlerOSX.m 42'></a> &nbsp;<b><a href='https://github.com/AzureAD/azure-activedirectory-library-for-objc/pull/309#discussion_r29541163'>File: ADAL-OSX/ADAL-OSX/ADNTLMHandlerOSX.m:L26-106</a></b>
- <a href='https://github.com/RPangrle-MS'></a> Add SAFE_ARC_RELEASE for _username and _password as well

<a href='https://www.codereviewhub.com/AzureAD/azure-activedirectory-library-for-objc/pull/309?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/AzureAD/azure-activedirectory-library-for-objc/pull/309?approve=1'></a>&nbsp;<a href='https://www.codereviewhub.com/AzureAD/azure-activedirectory-library-for-objc/pull/309?approve=0'></a>&nbsp;<a href='https://github.com/AzureAD/azure-activedirectory-library-for-objc/pull/309'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by brandwe** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/AzureAD/azure-activedirectory-library-for-objc/pull/309'>Refresh</a>.<a href='#crh-update-none'></a>

File: ADAL-OSX/ADAL-OSX/ADCredentialCollectionController.m [hunk 1]
```diff
@@ -34,11 +34,11 @@ -(id) init
...
-        SAFE_ARC_RETAIN(_customView);
-        SAFE_ARC_RETAIN(_usernameField);
-        SAFE_ARC_RETAIN(_passwordField);
-        SAFE_ARC_RETAIN(_usernameLabel);
-        SAFE_ARC_RETAIN(_passwordLabel);
+//        SAFE_ARC_RETAIN(_customView);
+//        SAFE_ARC_RETAIN(_usernameField);
+//        SAFE_ARC_RETAIN(_passwordField);
+//        SAFE_ARC_RETAIN(_usernameLabel);
+//        SAFE_ARC_RETAIN(_passwordLabel);
```
Comment (RPangrle-MS) [inline ADAL-OSX/ADAL-OSX/ADCredentialCollectionController.m]: Delete these lines entirely. We don't need them laying around.

File: ADAL-OSX/ADAL-OSX/ADNTLMHandlerOSX.m [hunk 2]
```diff
@@ -26,6 +26,80 @@
...
+    ADCredentialCollectionController *view = [ADCredentialCollectionController new];
+    [view.usernameLabel setStringValue:NSLocalizedString(@"User Name", nil)];
+    [view.passwordLabel setStringValue:NSLocalizedString(@"Password", nil)];
+    [_alert setAccessoryView:view.customView];
+    
+    [_alert beginSheetModalForWindow:[NSApp keyWindow] completionHandler:^(NSModalResponse returnCode) {
+        _response = returnCode;
+        
+        _username = [view.usernameField stringValue];
+        _password = [view.passwordField stringValue];
```
Comment (RPangrle-MS) [inline ADAL-OSX/ADAL-OSX/ADNTLMHandlerOSX.m]: Add some SAFE_ARC_RETAINs to these fields or use [self setUsername:] & [self setPassword:] (or the dot syntax)

File: ADAL-OSX/ADAL-OSX/ADNTLMHandlerOSX.m [hunk 3]
```diff
@@ -26,6 +26,80 @@
...
+
+    return self;
+}
+
+- (void)dealloc
+{
+    SAFE_ARC_RELEASE(_alert);
+    dispatch_release(_dsem);
+    SAFE_ARC_SUPER_DEALLOC();
+}
```
Comment (RPangrle-MS) [inline ADAL-OSX/ADAL-OSX/ADNTLMHandlerOSX.m]: Add SAFE_ARC_RELEASE for _username and _password as well

File: ADAL-OSX/ADAL-OSX/ADCredentialCollectionController.m (hunk 1 above)
Comment (kpanwar) [inline ADAL-OSX/ADAL-OSX/ADCredentialCollectionController.m]: :+1:

File: ADAL-OSX/ADAL-OSX/ADNTLMHandlerOSX.m (hunk 2 above)
Comment (kpanwar) [inline ADAL-OSX/ADAL-OSX/ADNTLMHandlerOSX.m]: :+1:<a href='#crh-update-none'></a>

File: ADAL-OSX/ADAL-OSX/ADNTLMHandlerOSX.m (hunk 3 above)
Comment (kpanwar) [inline ADAL-OSX/ADAL-OSX/ADNTLMHandlerOSX.m]: :+1:<a href='#crh-update-complete'></a>

**Comment by RPangrle-MS** [comment]
:shipit:

**Pull request merged**